### DISCUSSION TITLE
restarting TimerInterrupt on SAMD51 after wakeup

### DISCUSSION THREAD

**Discussion by DiversSi** [khoih-prog/SAMD_TimerInterrupt#26] [Q&A] [answered]
Hi,

how to use timerInterrupt when sending chip to deepsleap. I have problems restart the timer after wakeup of the chip.
Could not find suggestion anywhere...

**Comment by khoih-prog** [comment, marked as answer]
Sorry, I haven't used SAMD51 **deepsleep**.

Check if you can see something similar in

1. [**SAMD21 not waking up using Timer Interrupt from deep sleep**](https://forum.arduino.cc/t/samd21-not-waking-up-using-timer-interrupt-from-deep-sleep/662100)
2. [**SAMD21 wake up time from deep sleep**](https://forum.arduino.cc/t/samd21-wake-up-time-from-deep-sleep/593092)

You have to post your question on [**Arduino Forum**](https://forum.arduino.cc/) or [**Microchip Forum**](https://www.microchip.com/forums/f203.aspx) to ask for help, especially from SAMD expert [**MartinL**](https://forum.arduino.cc/u/MartinL).

Be sure to include all the necessary [**Minimal, Reproducible Example**](https://stackoverflow.com/help/minimal-reproducible-example) and follow the related Forum rules / etiquette. Or no one would be able to help you solve **this vague question**.

Good Luck,

**Reply by DiversSi** [reply]
Thanks for answer.

will try to post question to Arduino Forum..

Just for clarity ... what I wanted to use the TimerInterrupt for is creating 1 seccond triger for sending data to devices tft screen.
I send device to sleap using 
_DSB
_WFI
after trigering interrupt on external pin and waking up the the Device seem I can not restart the TimerInterrupt... 

Will let you know if I solve the problem,,,
Thanks for help!

**Comment by DiversSi** [comment]
Thanks for answer. Just for clarity, what I want to do is every 1 seccond sending to the tft updated data... for this I use your SAMDTimerInterrupt libraty.Than I send tge SAMD51 chip sleaping with _DC_WF After wake up I have problem restarting the TimerInterrupt.Will post question to Arduilo forum as you suggest to se if someone can help...Thanks for help,Igor PPoslano iz naprave Galaxy
-------- Izvirno sporočilo --------Od: Khoi Hoang ***@***.***> Datum: 18. 9. 22  17:43  (GMT+01:00) Za: khoih-prog/SAMD_TimerInterrupt ***@***.***> Kp: DiversSi ***@***.***>, Author ***@***.***> Zadeva: Re: [khoih-prog/SAMD_TimerInterrupt] restarting TimerInterrupt on
  SAMD51 after wakeup (Discussion #26) 
Sorry, I haven't used SAMD51 deepsleep.
Check if you can see something similar in

SAMD21 not waking up using Timer Interrupt from deep sleep
SAMD21 wake up time from deep sleep

You have to post your question on Arduino Forum or Microchip Forum to ask for help, especially from SAMD expert MartinL.
Be sure to include all the necessary Minimal, Reproducible Example and follow the related Forum rules / etiquette. Or no one would be able to help you solve this vague question.
Good Luck,
